api/users/serializers.py

```diff
@@ -13,18 +37,25 @@ class UserSerializer(JSONAPISerializer):
-    suffix = ser.CharField(help_text='For bibliographic citations')
+    given_name = ser.CharField(required=False, allow_blank=True, help_text='For bibliographic citations')
+    middle_name = ser.CharField(required=False, allow_blank=True, source='middle_names', help_text='For bibliographic citations')
+    family_name = ser.CharField(required=False, allow_blank=True, help_text='For bibliographic citations')
+    suffix = ser.CharField(required=False, allow_blank=True, help_text='For bibliographic citations')
     date_registered = ser.DateTimeField(read_only=True)
-    gravatar_url = ser.CharField(help_text='URL for the icon used to identify the user. Relies on http://gravatar.com ')
-    employment_institutions = ser.ListField(source='jobs', help_text='An array of dictionaries representing the '
-                                                                     'places the user has worked')
-    educational_institutions = ser.ListField(source='schools', help_text='An array of dictionaries representing the '
-                                                                         'places the user has attended school')
-    social_accounts = ser.DictField(source='social', help_text='A dictionary of various social media account '
-                                                               'identifiers including an array of user-defined URLs')
+    gravatar_url = ser.CharField(required=False, read_only=True, help_text='URL for the icon used to identify the user. Relies on http://gravatar.com ')
+    employment_institutions = JobsSerializer(read_only=True, required=False, source='jobs', help_text='An array of dictionaries representing the '
+                                                                                                      'places the user has worked')
+    educational_institutions = SchoolsSerializer(read_only=True, required=False, source='schools', help_text='An array of dictionaries representing the '
+                                                                                                             'places the user has worked')
+    # Social Fields are broken out to get around DRF complex object bug and to make API updating more user friendly.
+    gitHub = ser.CharField(required=False, source='social.github', allow_blank=True, help_text='Github Handle')
+    scholar = ser.CharField(required=False, source='social.scholar', allow_blank=True, help_text='Google Scholar Account')
+    personal_website = ser.CharField(required=False, source='social.personal', allow_blank=True, help_text='Personal Website')
+    twitter = ser.CharField(required=False, source='social.twitter', allow_blank=True, help_text='Twitter Handle')
+    linkedIn = ser.CharField(required=False, source='social.linkedIn', allow_blank=True, help_text='LinkedIn Account')
+    impactStory = ser.CharField(required=False, source='social.impactStory', allow_blank=True, help_text='ImpactStory Account')
```

Comment:
They should start with lowercase. Aside from that, if it's a brand name, they should follow the brand's casing and spacing from that point on, with spaces becoming underscores. If they are not brand names, snake_case.